eng/Publishing.props

```diff
@@ -1,5 +1,21 @@
 <Project>
   <PropertyGroup>
     <PublishingVersion>3</PublishingVersion>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);_PublishPackages</PublishDependsOnTargets>
```

Comment:
Why do we need a separate target to add to `ItemsToPushToBlobFeed`?

Reply:
I've followed the existing model we have in `aspnetcore` today. We also define `PublishDependsOnTargets` which is meant to provide this type of extensibility and seems cleaner, as targets run in order, items outside of target would be evaluated/defined before all other targets.

Reply:
We don't care about the order here as items that are defined as part of project evaluation are always present in any target run. We should really avoid any extra targets here in these extension points unless they are really necessary.

This whole project file could just be this (The `PublishingVersion` property isn't necessary anymore, it's just boilerplate):

```xml
<Project>

  <ItemGroup>
    <!--
      Additional packages needed to avoid prebuilts in VMR - https://github.com/dotnet/source-build/issues/4203
    -->
    <ItemsToPushToBlobFeed Include="$(ArtifactsPackagesDir)Dependency\**\*.nupkg"
                           IsShipping="true"
                           UploadPathSegment="Runtime"
                           Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
  </ItemGroup>

</Project>
```

Reply:
I've pushed the fix to my branch (https://github.com/NikolaMilosavljevic/fsharp/commit/10c6a33f3fd759dc67f8cae13235e1c50473f404) but for some reason it's not yet showing in this PR.

Reply:
Fixed with https://github.com/dotnet/fsharp/pull/16838/commits/10c6a33f3fd759dc67f8cae13235e1c50473f404